runners/core-java/src/main/java/org/apache/beam/runners/core/SimpleDoFnRunner.java

```diff
@@ -609,6 +724,31 @@ public TimeDomain timeDomain() {
       throw new UnsupportedOperationException("ProcessContext parameters are not supported.");
     }
 
+    @Override
+    public InputT element(DoFn<InputT, OutputT> doFn) {
+      throw new UnsupportedOperationException("Element parameters are not supported.");
+    }
+
+    @Override
+    public Instant timestamp(DoFn<InputT, OutputT> doFn) {
+      return timestamp();
+    }
+
+    @Override
+    public TimeDomain timeDomain(DoFn<InputT, OutputT> doFn) {
+      return timeDomain();
+    }
+
+    @Override
+    public OutputReceiver<OutputT> outputReceiver(DoFn<InputT, OutputT> doFn) {
+      return new WindowedContextOutputReceiver<>(this);
```

Comment:
Why not return `this`, it effectively implements `OutputReceiver` already.

Ditto elsewhere.